### PR TITLE
fixed a few more cppcheck and clang tidy warnings

### PR DESCRIPTION
**Pull request by michaeljbrazell**

### CODE REVIEW

File: amr-wind/core/Field.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@ FieldInfo::FieldInfo(
 {
     for (int i = 0; i < AMREX_SPACEDIM * 2; ++i) {
         m_bc_type[i] = BC::undefined;
+        m_bc_values_d[i] = nullptr;
```
Comment (sayerhs) [inline amr-wind/core/Field.cpp:31]: Would it be possible to do this in constructor initializer list?

**Review by sayerhs — APPROVED** [review]

**Review by jrood-nrel — APPROVED** [review]

File: amr-wind/core/Field.cpp (hunk 1 above)
Comment (michaeljbrazell) [reply amr-wind/core/Field.cpp:31]: like this?
`    , m_bc_values(AMREX_SPACEDIM * 2, amrex::Vector<amrex::Real>(ncomp, 0.0))
    , m_bc_values_d({})
    , m_bc_values_dview(ncomp * AMREX_SPACEDIM * 2)`

Comment (sayerhs) [reply amr-wind/core/Field.cpp:31]:
```
, m_bc_values_d(nullptr)
``` 

Since this is an AMReX data structure. I was curious if it had this option.

Comment (michaeljbrazell) [reply amr-wind/core/Field.cpp:31]: it doesn't have that option :(

Comment (sayerhs) [reply amr-wind/core/Field.cpp:31]: Thanks for clearing that up.

File: amr-wind/turbulence/LES/OneEqKsgs.cpp [hunk 2]
```diff
@@ -41,7 +41,7 @@ OneEqKsgsM84<Transport>::OneEqKsgsM84(CFDSim& sim)
     }
 
     {
-        const std::string coeffs_dict = this->model_name() + "_coeffs";
+        const std::string coeffs_dict = model_name() + "_coeffs";
```
Comment (sayerhs) [inline amr-wind/turbulence/LES/OneEqKsgs.cpp:44]: I doubt this will fix the issue... as the method is still declared virtual.

Comment (michaeljbrazell) [reply amr-wind/turbulence/LES/OneEqKsgs.cpp:44]: darn, I need to set up clang-tidy on my machine instead of depending on CI

Comment (jrood-nrel) [reply amr-wind/turbulence/LES/OneEqKsgs.cpp:44]: You can use it on eagle with `module load llvm` or rhodes when eagle is down.

**Pull request merged**